src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/MetadataName.T.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Threading.RateLimiting
+{
+    /// <summary>
+    /// A strongly-typed name of metadata that can be stored in a <see cref="RateLimitLease"/>.
+    /// </summary>
+    /// <typeparam name="T">The type the metadata will be.</typeparam>
+    public sealed class MetadataName<T> : IEquatable<MetadataName<T>>
+    {
+        private readonly string _name;
+
+        /// <summary>
+        /// Constructs a <see cref="MetadataName{T}"/> object with the given name.
+        /// </summary>
+        /// <param name="name">The name of the <see cref="MetadataName"/> object.</param>
+        public MetadataName(string name)
+        {
+            _name = name;
+        }
+
+        /// <summary>
+        /// Gets the name of the metadata.
+        /// </summary>
+        public string Name => _name;
+
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            return _name ?? string.Empty;
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            return _name == null ? 0 : _name.GetHashCode();
+        }
+
+        /// <inheritdoc/>
+        public override bool Equals([NotNullWhen(true)] object? obj)
+        {
+            return obj is MetadataName<T> && Equals((MetadataName<T>)obj);
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(MetadataName<T>? other)
```

Comment:
Added `?` since `IEquatable` requires it, and added the `if (other is null)` check